src/search/cost.cc

```diff
@@ -192,9 +194,291 @@ bool is_win_legal(inst* orig, int len1, inst* synth, int len2,
+      //   cout << "win prog check: " << is_equal << " " << is_equal_expected << endl;
+      //   if (is_equal_expected != is_equal) {
+      //     cout << "win prog check fail: " << is_equal << " " << is_equal_expected << endl;
+      //   }
+      //   smt_var::is_win = true;
+      // }
+      pass_validator = true;
+    } catch (const string err_msg) {
+      // illegal program
+      //synth->set_error_cost(ERROR_COST_MAX);
+      //synth->set_safety_cost(ERROR_COST_MAX);
+      //return ERROR_COST_MAX;
+      pass_validator = false;
+    }
+    auto t2 = NOW;
+    auto dur = DUR(t1, t2);
+    // cout << dur << endl;
+    dur_sum += dur;
+    if (dur > 50000) {
+      dur_sum_long += dur;
+      n_sum_long++;
+      // synth->print();
+    }
+  }
+  
+
+  //int avg_value = get_avg_value(ex_set_size);
+  //total_cost = get_final_error_cost(total_cost, is_equal,
+                                    //ex_set_size, num_successful_ex,
+                                    //avg_value);
+  // process counterexamples
+  // If num_successful_ex < (int)_examples._exs.size(),
+  // it shows the example that synth fails in the example set is a counterexample.
+  // The counterexample generated from this synth may have already been in the examples set.
+  // Thus, only when num_successful_ex == (int)_examples._exs.size(),
+  // the counterexample generated from this synth must can be added into the example set.
+  // But it should ensure that the number of initial example set is big enough.
+  // case 1: gen_counterex_flag = (is_equal == 0);
+  // case 2: gen_counterex_flag = (is_equal == 0) && (num_successful_ex == (int)_examples._exs.size());
+  
+  //CONDITION: ONLY IF ITS UNSAFE BUT PASSES FIRST 2 MODULES
+  //VALIDATOR GENERATES COUNTEREXAMPLE 
+  //if (((is_equal == 0) || (is_equal == ILLEGAL_CEX)) &&
+  //    (num_successful_ex == (int)_examples._exs.size())) {
+  if(pass_static_safety && (num_of_unsucc_ex == 0) && ((pass_validator == false)||is_equal==-1)){
+    _examples.insert(_vld._last_counterex);
+    _meas_new_counterex_gened = true;
+    cout << "counterexample " << _examples.size() << ":" << endl;
+    cout << _vld._last_counterex.input << endl;
+    cout << _vld._last_counterex.output << endl;
+  }
+  // in case there is overflow which makes a positive value become a negative value or
+  // total_cost > ERROR_COST_MAX
+  // if ((total_cost > ERROR_COST_MAX) || (total_cost < 0)) {
+  //   synth->set_error_cost(ERROR_COST_MAX);
+  //   return ERROR_COST_MAX;
+  // }
+
+  //calculate safety cost based on the 3 variables
+  //think of some good equation
+  //set_safety_cost below:
+  //synth->set_error_cost(total_cost);
+  //return total_cost;
+  //Some function of: pass_satic_safety, num_of_unsucc_ex, pass_validator, isEqual
+
+  //The below is a very rough calculation as a placeholder for now
+  //Later the safety cost will be made more continuous and accurate
```

Comment:
It's good code hygiene to avoid long lines of comments unless you're documenting some important behavior of the code. Also use small letters, using capital letters makes it sound like one is shouting :-)